lib/uplinkc/testdata/object_test.c

```diff
@@ -187,6 +187,36 @@ void handle_project(ProjectRef project) {
             free_downloader(downloader);
         }
 
+        { // download range
+            long start = 100;
+            long limit = 1024 * 4;
+            DownloaderRef downloader = download_range(bucket, object_paths[i], start, limit, err);
+            require_noerror(*err);
+
+            uint8_t *downloaded_data = malloc(data_len);
```

Comment:
My understanding is that we only need to allocate memory for limit; if we read more something isn't right and we'd get a panic.